src/mapping/cellranger_multi/script.py

```diff
@@ -140,19 +181,36 @@ def check_subset_dict_equal_length(group_name: str,
     assert len(set(lens.values())) <= 1, f"The number of values passed to {group_name} "\
                                          f"arguments must be 0, 1 or all the same. Offenders: {lens}"
 
-def process_params(par: dict[str, Any]) -> str:
-    # if par_input is a directory, look for fastq files
-    par["input"] = [Path(fastq) for fastq in par["input"]]
-    if len(par["input"]) == 1 and par["input"][0].is_dir():
-        logger.info("Detected '--input' as a directory, "
+def resolve_input_directories_to_fastq_paths(input_paths: list[str]) -> list[Path]:
+
+    input_paths = [Path(fastq) for fastq in input_paths]
+    if len(input_paths) == 1 and input_paths[0].is_dir():
+        logger.info("Detected a directory in input paths, "
                     "traversing to see if we can detect any FASTQ files.")
-        par["input"] = [input_path for input_path in par["input"][0].rglob('*')
+        input_paths = [input_path for input_path in input_paths[0].rglob('*')
                         if re.match(fastq_regex, input_path.name) ]
 
     # check input fastq files
-    for input_path in par["input"]:
+    for input_path in input_paths:
         assert re.match(fastq_regex, input_path.name) is not None, \
```

Comment:
Do we check somewhere that these paths are actually files (and not directories for example?)

Reply:
> Do we check somewhere that these paths are actually files (and not directories for example?)

Yes, both input and helper input are resolved to files if a directory is provided (https://github.com/openpipelines-bio/openpipeline/blob/0792f58f8ce7a9044b456822f67e7d7868860d41/src/mapping/cellranger_multi/script.py#L143)